lib/migration-stream.js

```diff
@@ -36,37 +36,60 @@ function MigrationStream() {
 
     var info = mapnik.VectorTile.info(tile.buffer);
 
-    if (info.errors) {
-      var err;
-      if (info.tile_errors) {
-        err = new Error(info.tile_errors[0].replace(' message', ''));
-      } else if (info.layers) {
-        for (var i = 0; i < info.layers.length; i++) {
-          var layer = info.layers[i];
-          if (layer.errors) {
-            err = new Error(layer.errors[0].replace(' message', ''));
-            break;
-          }
-        }
-      } else {
-        err = new Error('Invalid data');
-      }
-      err.code = 'EINVALID';
-      return callback(err);
-    }
-
     var v2 = info.layers.every(function(layer) {
         return layer.version === 2;
     });
 
     if (v2 === true) {
+      // Check for errors on the v2 tile
+      if (info.errors) {
+        var err;
+        if (info.tile_errors) {
+          err = new Error(info.tile_errors[0].replace(' message', ''));
+        } else if (info.layers) {
+          for (var i = 0; i < info.layers.length; i++) {
+            var layer = info.layers[i];
+            if (layer.errors) {
+              err = new Error(layer.errors[0].replace(' message', ''));
+              break;
+            }
+          }
+        } else {
+          err = new Error('Invalid data');
+        }
+        err.code = 'EINVALID';
+        return callback(err);
+      }
+
       migrationStream.push(tile);
       return callback();
     }
 
 
     migrate(tile, function(err, tile) {
       if (err) return callback(err);
+
+      // Check for errors on newly-minted v2 tile
+      var info = mapnik.VectorTile.info(tile.buffer);
+      if (info.errors) {
+        var err;
+        if (info.tile_errors) {
+          err = new Error(info.tile_errors[0].replace(' message', ''));
+        } else if (info.layers) {
+          for (var i = 0; i < info.layers.length; i++) {
+            var layer = info.layers[i];
+            if (layer.errors) {
+              err = new Error(layer.errors[0].replace(' message', ''));
+              break;
+            }
+          }
+        } else {
+          err = new Error('Invalid data');
+        }
+        err.code = 'EINVALID';
+        return callback(err);
+      }
+
```

Comment:
@jakepruitt wondering if we could turn this `if(info.errors)` check into a function now that we are using it twice?